### PR TITLE
Refactor handling of external resources

### PR DESCRIPTION
**Pull request by fcollot**
The previous `add_external_resources` cmake command is replaced with two commands `set_external_resources` and `import_external_resources`:
- `set_external_resources` is used to declare that a target "provides" external resources.
- `import_external_resources` is used to import resources into the resource directory of a target (this command can be used alone if the resources are provided by the same target).

The main goal of this two-step process is that there are targets that will provide or generate resources (such as Python modules), and these resources will then be imported into the executable target (which, on macOS, will place them in the application bundle) through post-build commands.

Another important change is the possibility to place resources in subdirectories.

### CODE REVIEW

**Comment by fcollot** [comment]
(Please note that medPython will not compile with these changes. The next PR's will adapt it so the they will have to be merged together)

File: src/cmake/module/set_external_resources.cmake [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+#
+# APPEND
+#     With this option, the resources are added to the previous resources of the
+#     target. Leaving the option out will drop the previously declared
+#     resources.
+#
+# DESTINATION
+#     Specify a subdirectory relative to the final import location.
+#
+# FILES, DIERCTORIES
```
Comment (mathildemerle) [inline src/cmake/module/set_external_resources.cmake:41]: DIRECTORIES

**Review by mathildemerle — APPROVED** [review]

File: src/cmake/module/import_external_resources.cmake [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    endwhile()
+
+endmacro()
+
+macro(_get_resources_parent_directory parent_dir_var target)
+
+    set(${parent_dir_var} $<IF:$<PLATFORM_ID:Darwin>,$<TARGET_BUNDLE_CONTENT_DIR:${target}>,${CMAKE_BINARY_DIR}>)
+    get_property(is_multi_config GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+    if (${is_multi_config})
+        set(config_subdir /$<$<CONFIG:Debug>:Debug>$<$<CONFIG:Release>:Release>$<$<CONFIG:RelWithDebInfo>:RelWithDebInfo>$<$<CONFIG:MinSizeRel>:MinSizeRel>)
```
[the remark this answers is not in the document]
Comment (fcollot) [reply src/cmake/module/import_external_resources.cmake:165]: These lines are necessary for multi-configuration generators (such as Visual Studio), and I can't do much about the syntax cmake has chosen for generator expressions.

[the remark this answers is not in the document]
Comment (mathildemerle) [reply src/cmake/module/import_external_resources.cmake:165]: indeed, no problem if it's needed ^^

File: src/cmake/module/import_external_resources.cmake [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            list(APPEND ${current_var} ${item})
+        endif()
+        list(POP_FRONT ${resources_var} item)
+    endwhile()
+
+endmacro()
+
+macro(_get_resources_parent_directory parent_dir_var target)
+
+    set(${parent_dir_var} $<IF:$<PLATFORM_ID:Darwin>,$<TARGET_BUNDLE_CONTENT_DIR:${target}>,${CMAKE_BINARY_DIR}>)
```
[the remark this answers is not in the document]
Comment (fcollot) [reply src/cmake/module/import_external_resources.cmake:162]: The platform identifier for macOS (Darwin is the base OS on which macOS is built).

**Comment by fcollot** [comment]
New changes:
- fix DIRECTORIES spelling error.
- remove platform test from generator expressions because in the example below the TARGET_BUNDLE_CONTENT_DIR expression is evaluated even if it is not used, which causes an error on non-apple platforms.
  `$<IF:$<PLATFORM_ID:Darwin>,$<TARGET_BUNDLE_CONTENT_DIR:${target}>,${CMAKE_BINARY_DIR}>`

**Review by mathildemerle — APPROVED** [review]

**Review by Florent2305 — APPROVED** [review]
ok

**Pull request merged**